### DISCUSSION TITLE
Has Many [Aug 31]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#3735] [Discord Threads]
> 🙋‍♂️ **Eric**  
#### this [github issue](https://github.com/nocodb/nocodb/issues/525#issue-993335776) seems to show the exact design flow, but in all recent tags of the source this datatype is commented out.  
![image0](https://cdn.discordapp.com/attachments/661905455894888493/1011384717217632378/unknown.png)

**Comment by o1lab** [comment]
> 💁 **o1lab**  
#### Has Many is One to Many  
#  
> 🙋‍♂️ **Eric**  
#### well conceptually, but it also bakes in exclusivity from the perspective of the parent.  
#  
> 🙋‍♂️ **Eric**  
#### e.g.   if i define one to many in the current interface, if i assign a link value from the parent, it removes that from other peers.   so not sure if it's just a bug/quirk in the current implementation.  
#  
> 🙋‍♂️ **Eric**  
#### e.g.  
[firefox_6j1ayhc7Xf.mp4](https://cdn.discordapp.com/attachments/1011384718060695704/1011409681618575520/firefox_6j1ayhc7Xf.mp4)  
#  
> 🙋‍♂️ **Russell**  
#### A **Device** can have many **current_cal**, but a **current_cal** can only have one **Device**. In the **current_cal** table there should be a column for **Devices_id** which only holds an integer ID. So if you assign *“this is a yaml 1”* to Device ID 2, it will no longer be assigned to Device ID 1. So it seems like it’s working as expected, or do you want many-to-many, or did you want one-to-many in the opposite direction?  
#  
> 🙋‍♂️ **Eric**  
#### The desire is to have a conventional foreign key,   in the context desribed here, yes running in the opposite direction... a foreign key from one table to another would not generally impart or enforce any exclusivity (across) the members of the parent table.  
#  
> 🙋‍♂️ **Eric**  
#### e.g.   one should be able to (re) use the foreign keys values; many parent entities having the same foreign key value.  
#  
> 🙋‍♂️ **Eric**  
#### from looking at the docs it appeared that this exists, but maybe rolled into the linked  
#  
> 🙋‍♂️ **Eric**  
#### <@531097245194387457>  it looks like we have the same question/need.  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1011384718060695704)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
